pkg/ebpf/netlink/socket_buffer.go

```diff
@@ -0,0 +1,45 @@
+package netlink
+
+import (
+	"syscall"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/mdlayher/netlink"
+)
+
+// setSocketBufferSize of a netlink socket.
+// We use this custom function as oposed to netlink.Conn.SetReadBuffer because you can only
+// set a value higher than /proc/sys/net/core/rmem_default (which is around 200kb for most systems)
+// if you use SO_RCVBUFFORCE with CAP_NET_ADMIN (https://linux.die.net/man/7/socket).
+func setSocketBufferSize(sizeInBytes int, c *netlink.Conn) {
+	rawConn, err := c.SyscallConn()
+	if err != nil {
+		log.Warnf("error obtaining raw_conn %s", err)
```

Comment:
Should the `Warnf` be `Errorf` since the message talk of "error" ?

Also should we not instead return a error and print it from the calling function ?